scripts/generate_linkml_docs.py

```diff
@@ -7,42 +7,50 @@
 from tempfile import TemporaryDirectory
 from typer import run
 from linkml.generators.markdowngen import MarkdownGenerator
+from script_utils.cli import echo_failure, echo_success
 
 HERE = Path(__file__).parent.resolve()
 LINKML_YAML = HERE.parent / "src" / "schema" / "submission_centric_schema.yaml"
 DOCS_DIR = HERE.parent / "docs" / "schema_markdown"
 
 
-def generate_linkml_markdown(docs_dir):
+def generate_linkml_markdown(docs_dir: Path):
     """Function to generate markdown documentations of a given schema"""
     return MarkdownGenerator(schema=str(LINKML_YAML), directory_output=True).serialize(
         directory=docs_dir
     )
 
 
-def compare_folders(expected: str, observed: str):
+def compare_folders(expected: Path, observed: Path):
     """Function to check equality of contents of two folders"""
     try:
-        subprocess.run(
-            ["diff", "-arq", expected, observed],
+        process = subprocess.run(
+            ["diff", "-arq", str(expected), str(observed)],
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE,
             check=True,
         )
+        return process.returncode, process.stdout
     except subprocess.CalledProcessError as exc:
-        print("Status : FAIL", exc.returncode, exc.stdout)
-        sys.exit(1)
+        return exc.returncode, exc.stdout
```

Comment:
This is exposing implementation details of the compare_folders functions. Other functions should not know that it is implemented using a system call and thus have exit code and standard out. It would be better to raise an exception.